firmware/main.c

```diff
@@ -827,23 +821,35 @@ static void cmd_motor_linear(BaseSequentialStream *chp, int argc, char *argv[])
 	palSetPadMode(pinPorts[pinEnable].gpio, pinPorts[pinEnable].pin, PAL_MODE_OUTPUT_PUSHPULL);
 	palClearPad(pinPorts[pinEnable].gpio, pinPorts[pinEnable].pin); // Reset pins as its default state is high
 
+	char direction_literal[25];
```

Comment:
make it a char*, as above. with constant strings just assigned to it on demand.
init to NULL.